canaryconfigmgr/canaryConfigMgr.go

```diff
@@ -266,23 +344,21 @@ func (canaryCfgMgr *canaryConfigMgr) updateHttpTriggerWithRetries(triggerName, t
 	for i := 0; i < fission.MaxRetries; i++ {
 		triggerObj, err := canaryCfgMgr.fissionClient.HTTPTriggers(triggerNamespace).Get(triggerName)
 		if err != nil {
-			log.Printf("Error getting http trigger object : %v", err)
-			return err
+			return errors.Wrap(err, "error getting http trigger object")
 		}
 
 		triggerObj.Spec.FunctionReference.FunctionWeights = fnWeights
 
 		_, err = canaryCfgMgr.fissionClient.HTTPTriggers(triggerNamespace).Update(triggerObj)
 		switch {
 		case err == nil:
-			log.Printf("Updated Http trigger : %s.%s", triggerName, triggerNamespace)
+			canaryCfgMgr.logger.Info("updated http trigger", zap.String("trigger", fmt.Sprintf("%s.%s", triggerName, triggerNamespace)))
 			return nil
 		case k8serrors.IsConflict(err):
-			log.Printf("Conflict in updating http trigger : %s.%s, retrying", triggerName, triggerNamespace)
+			canaryCfgMgr.logger.Info("conflict in updating http trigger, retrying", zap.Error(err), zap.String("trigger", fmt.Sprintf("%s.%s", triggerName, triggerNamespace)))
 			continue
 		default:
-			log.Printf("Error updating trigger : %s.%s = %v", triggerName, triggerNamespace, err)
-			return err
+			return errors.Wrapf(err, "error updating trigger: %s.%s", triggerName, triggerNamespace)
```

Comment:
log removed, is it intentional?

Reply:
nope, I'll add that back in - the thinking was this function was returning an error that would be logged at a higher level, but that's not the case.